instrumentation/runtime/test/runtime_test.go

```diff
@@ -0,0 +1,161 @@
+				Unit:        "By",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: false,
+					DataPoints:  []metricdata.DataPoint[int64]{{}},
+				},
+			},
+			{
+				Name:        "go.memory.allocated",
+				Description: "Memory allocated to the heap by the application.",
+				Unit:        "By",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: true,
+					DataPoints:  []metricdata.DataPoint[int64]{{}},
+				},
+			},
+			{
+				Name:        "go.memory.allocations",
+				Description: "Count of allocations to the heap by the application.",
+				Unit:        "{allocation}",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: true,
+					DataPoints:  []metricdata.DataPoint[int64]{{}},
+				},
+			},
+			{
+				Name:        "go.memory.gc.goal",
+				Description: "Heap size target for the end of the GC cycle.",
+				Unit:        "By",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: false,
+					DataPoints:  []metricdata.DataPoint[int64]{{}},
+				},
+			},
+			{
+				Name:        "go.goroutine.count",
+				Description: "Count of live goroutines.",
+				Unit:        "{goroutine}",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: false,
+					DataPoints:  []metricdata.DataPoint[int64]{{}},
+				},
+			},
+			{
+				Name:        "go.processor.limit",
+				Description: "The number of OS threads that can execute user-level Go code simultaneously.",
+				Unit:        "{thread}",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: false,
+					DataPoints:  []metricdata.DataPoint[int64]{{}},
+				},
+			},
+			{
+				Name:        "go.config.gogc",
+				Description: "Heap size target percentage configured by the user, otherwise 100.",
+				Unit:        "%",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: false,
+					DataPoints:  []metricdata.DataPoint[int64]{{}},
+				},
+			},
+		},
+	}
+	metricdatatest.AssertEqual(t, expectedScopeMetric, rm.ScopeMetrics[0], metricdatatest.IgnoreTimestamp(), metricdatatest.IgnoreValue())
+	assertNonZeroValues(t, rm.ScopeMetrics[0])
+}
+
+func assertNonZeroValues(t *testing.T, sm metricdata.ScopeMetrics) {
+	for _, m := range sm.Metrics {
+		switch a := m.Data.(type) {
+		case metricdata.Sum[int64]:
+			for _, dp := range a.DataPoints {
+				assert.True(t, dp.Value > 0, fmt.Sprintf("Metric %q should have a non-zero value for point with attributes %+v", m.Name, dp.Attributes))
+			}
+		default:
+			t.Fatalf("unexpected data type %v", a)
+		}
+	}
+}
+
+func allocateMemory(buffer [][]byte) {
+	newBuffer := make([]byte, 1000000)
+	for i := range newBuffer {
+		newBuffer[i] = 0
+	}
+	buffer = append(buffer, newBuffer)
+}
```

Comment:
The passed slice is never used again. The GC will collect the allocated space right after this returns, right?

Maybe pass a pointer, return the slice, or just replace this call with `buffer append(buffer, make([]byte, 1000000))`

Reply:
returned the slice.